### PR TITLE
Override builder commands for all builders based on repo and platform.

### PR DESCRIPTION
**Pull request by godofredoc**
This is a follow up to remove the 4m latency to select a bot while
waiting for the builder cache to be populated.

Bug: https://github.com/flutter/flutter/issues/84295

### CODE REVIEW

File: config/devicelab_config.star [hunk 1]
```diff
@@ -53,6 +55,8 @@ MAC_ANDROID_DEFAULT_CACHES = [
     swarming.cache(name = "flutter_sdk", path = "flutter sdk"),
     # Gradle cache
     swarming.cache(name = "gradle", path = "gradle"),
+    # Builder cache
+    swarming.cache(name = "builder_mac_devicelab", path = "builder"),
```
Comment (keyonghan) [inline config/devicelab_config.star:59]: Nit: builder_mac_android_devicelab

File: config/devicelab_staging_config.star [hunk 2]
```diff
@@ -43,6 +45,8 @@ MAC_ANDROID_DEFAULT_CACHES = [
     swarming.cache(name = "pub_cache", path = ".pub-cache"),
     # Flutter SDK code
     swarming.cache(name = "flutter_sdk", path = "flutter sdk"),
+    # Builder
+    swarming.cache(name = "builder_mac_devicelab_staging", path = "builder"),
```
Comment (keyonghan) [inline config/devicelab_staging_config.star:49]: nit: builder_mac_android_devicelab_staging.

**Review by keyonghan — APPROVED** [review]
LGTM with nits.

File: config/devicelab_config.star (hunk 1 above)
Comment (godofredoc) [reply config/devicelab_config.star:59]: We don't need to separate the cache per device type only per host type.

File: config/devicelab_staging_config.star (hunk 2 above)
Comment (godofredoc) [reply config/devicelab_staging_config.star:49]: Same answer as above.

**Pull request merged**